autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/handlers/PopUpFoundMessageHandlerPlugin.kt

```diff
@@ -17,24 +17,53 @@
 package com.duckduckgo.autoconsent.impl.handlers
 
 import android.webkit.WebView
+import com.duckduckgo.app.global.DispatcherProvider
 import com.duckduckgo.autoconsent.api.AutoconsentCallback
 import com.duckduckgo.autoconsent.impl.MessageHandlerPlugin
+import com.duckduckgo.autoconsent.impl.adapters.JSONObjectAdapter
 import com.duckduckgo.autoconsent.store.AutoconsentSettingsRepository
 import com.duckduckgo.di.scopes.AppScope
 import com.squareup.anvil.annotations.ContributesMultibinding
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import kotlinx.coroutines.runBlocking
+import timber.log.Timber
 import javax.inject.Inject
 
 @ContributesMultibinding(AppScope::class)
-class PopUpFoundMessageHandlerPlugin @Inject constructor(private val repository: AutoconsentSettingsRepository) : MessageHandlerPlugin {
+class PopUpFoundMessageHandlerPlugin @Inject constructor(
+    private val dispatcherProvider: DispatcherProvider,
+    private val repository: AutoconsentSettingsRepository,
+) : MessageHandlerPlugin {
+
+    private val moshi = Moshi.Builder().add(JSONObjectAdapter()).build()
 
     override fun process(messageType: String, jsonString: String, webView: WebView, autoconsentCallback: AutoconsentCallback) {
-        if (supportedTypes.contains(messageType)) {
-            if (repository.userSetting) {
-                return
+        try {
+            if (supportedTypes.contains(messageType)) {
+                val message: PopUpFoundMessage = parseMessage(jsonString) ?: return
+                if (repository.userSetting) return
+
+                val url: String = runBlocking(dispatcherProvider.main()) { webView.url.orEmpty() }
+                if (message.cmp.endsWith(IGNORE_CMP_SUFFIX)) return
+                if (message.url != url) return
```

Comment:
You sure we need this check? Aren't we sure at this point that the message originated from `webView`?

Reply:
Probably not but I've seen in a few cases that moving quickly from one site with a cmp to another one with a cmp you might get the first popupFound message triggered while you are on the second site. The check is to be sure that both the url from the message and webview are the same before continuing.

For the record, I saw this when I moved the injecting to onPageFinished so I might be overdoing it here and I don't think I've seen that scenario before.

Reply:
It should be fine as long as `webView.url` returns the same as `location.href` in JS. If it gets out of sync or differs (e.g. if the #anchor part is missing), we could miss things here

Reply:
Yeah I ran a few tests and seemed to work fine but let's remove it just in case, like I said I haven't seen this again since moving the injecting back to onPageStarted so we can discuss if I can recreate again at some point.